### ISSUE TITLE
Advised way to do DR server dcs?

### ISSUE THREAD

**Issue by infa-bsurber** [hashicorp/consul-helm#197]
I'm currently running a Vault backed by a Consul cluster deployed by this chart (notably, including bootstrapACLs enabled) and I'm looking to setup a DR Consul cluster in another region.

What's the advised way to use this chart to deploy to another region, set it as a new datacenter, and have it join the first cluster?

I think I can use consul-replicate to do a master-follower KV replication to the DR cluster but the DR cluster has to be joined via WAN.

**Comment by jtcressy** [comment]
Bump on this. (plus some more info from my own investigation)

I setup 3 kubernetes clusters in different regions via GKE and ensured the pods could talk to each other via vpc-native network. I deployed this helm chart and set the datacenter on each to "<clustername>-<region-name>". e.g. I had `test1-us-central1`, `test2-us-east1`, and `test3-us-east4`. I connected to one of the test2-us-east1 server nodes by `kubectl port-forward svc/consul-server 8500:8500`  and ran `consul join -token=<bootstrap acl token for test2-us-east1> -wan <ip of server pod in test1-us-central1>`. Repeated once more for one of the other servers in test3-us-east4.

This network setup is unique to GCP because a VPC can be globally-routable with a /20 in each region. i.e. an instance in us-east1 can directly ping an instance in us-central1 etc. 
On other cloud providers it gets more complicated because you have to dig into huge mesh-vpn infra in order to have cross-region interconnect.

I also investigated using ACL token replication and found it a very difficult task to complete manually with this helm chart. It involved appending info to server and client configmaps, and manually creating ACL policies plus an ACL replication token in the primary DC that I had to distribute among the secondary DC's. More info on this process here: https://learn.hashicorp.com/consul/day-2-operations/acl-replication

Likewise, I imagine the process of setting up consul-replicate to clone KV values would be more tedious. However if you have Vault Enterprise Premium, you can just make use of Vault DR Replication and keep your consul KV namespaces segregated to their respective DC's. (for my purposes, I plan to use Vault DR instead of replicating consul KV)

In short, this chart has no simple way of implementing the multi-datacenter features of consul, and I might even file a separate feature-request issue to begin conversations around implementing better multi-dc support.

**Comment by david-yu** [comment]
Hi there, we now have a documented way to perform DR from one region to another for WAN Federated Datacenters for Consul Kubernetes: https://learn.hashicorp.com/tutorials/consul/kubernetes-disaster-recovery?in=consul/kubernetes-production
